### ISSUE TITLE
Inconsistency in Unit windows

### ISSUE THREAD

**Issue by bevans2000** [mars-sim/mars-sim#531]
**Describe the issue**
There is a lot of duplicated and repeated logic throughout the various  TabPanels. This also leads to different UI styles in some places.

**Expected behavior**
Consolidate and standardise the common visual elements and logic into the TabPanel class. This will reduce some of the code quality warnings concerning duplicate code whilst making the individual TabPanel classes more focused on displaying the specific content at hand.
Some rationalisation has already been done as part of the BuildingUnitWindow.

**Comment by mokun** [comment]
Sure. It makes sense to unify how each tab panel class should look and behave.

**Comment by bevans2000** [comment]
I'm going to look at the use of 'uiPulse' method as well. I found a big inefficiency with the tick counter being shared by all listeners so basically the UI pulse fires a lot more than expected. We should only have a single pulse method but controls how often it fires relative to the clock.
I will create a pull request soon with the reworked ui panels. I have removed a lot if code and replaced it with shared code. There are a couple layout adnormalites that need flushing out.
